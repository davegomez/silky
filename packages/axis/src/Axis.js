// @flow
import React from 'react';
import type { CSS } from '../../utils/types';
import { Group } from '@silky/group';

type Props = {
  className?: string,
  css?: CSS,
  position: string,
};

export default function Text({ className = '', css, position }: Props) {
  return (
    <Group className={`silky-axis axis-${position} ${className}`} css={css}>
      {/* Test */}
    </Group>
  );
}
