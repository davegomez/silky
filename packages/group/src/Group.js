// @flow
import React from 'react';
import { getTransform } from '../../utils';
import type { CSS, Transform } from '../../utils/types';

type Props = {
  children: any,
  className?: string,
  css?: CSS,
  transform?: Transform,
};

export default function Group({
  children,
  className = '',
  css,
  transform,
  ...restProps
}: Props) {
  return (
    <g
      className={`silky-group ${className}`}
      css={css}
      transform={getTransform(transform)}
      {...restProps}
    >
      {children}
    </g>
  );
}
