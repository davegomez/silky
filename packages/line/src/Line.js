// @flow
import React from 'react';
import type { CSS } from '../../utils/types';

type Props = {
  className?: string,
  css?: CSS,
  x1: number | string,
  x2?: number | string,
  y1: number | string,
  y2?: number | string,
  pathLength?: number,
};

export default function Line({
  className = '',
  css,
  x1,
  x2,
  y1,
  y2,
  pathLength,
  ...restProps
}: Props) {
  return (
    <line
      className={`silky-line ${className}`}
      css={css}
      x1={x1}
      x2={x2}
      y1={y1}
      y2={y2}
      pathLength={pathLength}
      {...restProps}
    />
  );
}
