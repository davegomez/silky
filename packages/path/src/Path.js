// @flow
import React from 'react';
import type { CSS } from '../../utils/types';

type Props = {
  className?: string,
  css?: CSS,
  d: string,
  pathLength?: number,
};

export default function Path({
  className = '',
  css,
  d,
  pathLength,
  ...restProps
}: Props) {
  return (
    <path
      className={`silky-path ${className}`}
      css={css}
      d={d}
      pathLength={pathLength}
      {...restProps}
    />
  );
}
