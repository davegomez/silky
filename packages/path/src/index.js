export { default as Path } from './Path';
