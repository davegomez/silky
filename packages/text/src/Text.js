// @flow
import React from 'react';
import { getTransform } from '../../utils';
import type { CSS, Transform } from '../../utils/types';

type Props = {
  children: any,
  className?: string,
  css?: CSS,
  dx?: number | string,
  dy?: number | string,
  lengthAdjust?: 'spacing' | 'spacingAndGlyphs',
  transform?: Transform,
  textLength?: number | string,
  x?: number | string,
  y?: number | string,
};

export default function Text({
  children,
  className = '',
  css,
  dx,
  dy,
  lengthAdjust,
  transform,
  textLength,
  x,
  y,
  ...restProps
}: Props) {
  return (
    <text
      className={`silky-text ${className}`}
      css={css}
      dx={dx}
      dy={dy}
      lengthAdjust={lengthAdjust}
      textLength={textLength}
      transform={getTransform(transform)}
      x={x}
      y={y}
      {...restProps}
    >
      {children}
    </text>
  );
}
