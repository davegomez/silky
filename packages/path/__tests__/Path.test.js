import React from 'react';
import serializer, { matchers } from 'jest-emotion';
import { render, cleanup } from 'react-testing-library';
import { css } from '@emotion/core';
import { Path } from '../src';

expect.addSnapshotSerializer(serializer);
expect.extend(matchers);
afterEach(cleanup);

const d =
  'M 10,30 A 20,20 0,0,1 50,30 A 20,20 0,0,1 90,30 Q 90,60 50,90 Q 10,60 10,30 z';

test('Path', () => {
  const { container } = render(
    <svg>
      <Path d={d} />
    </svg>
  );

  const path = container.querySelector('path');
  expect(path).toMatchSnapshot();
  expect(path.classList.contains('silky-path')).toBe(true);
  expect(path.getAttribute('d')).toBe(d);
});

test('Path with custom class', () => {
  const { container } = render(
    <svg>
      <Path className="foo" d={d} />
    </svg>
  );

  const path = container.querySelector('path');
  expect(path).toMatchSnapshot();
  expect(path.classList.contains('silky-path')).toBe(true);
  expect(path.classList.contains('foo')).toBe(true);
});

test('Path with Emotion', () => {
  const styles = css`
    color: black;
  `;
  const { container } = render(
    <svg>
      <Path css={styles} d={d} />
    </svg>
  );

  const path = container.querySelector('path');
  expect(path).toMatchSnapshot();
  expect(path.classList.contains('silky-path')).toBe(true);
  expect(path).toHaveStyleRule('color', 'black');
});

test('Path with pathLength', () => {
  const { container } = render(
    <svg>
      <Path d={d} pathLength={2} />
    </svg>
  );

  const path = container.querySelector('path');
  expect(path).toMatchSnapshot();
  expect(path.classList.contains('silky-path')).toBe(true);
  expect(path.getAttribute('pathLength')).toBe('2');
});
