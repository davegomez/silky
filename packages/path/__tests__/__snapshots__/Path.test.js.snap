// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Path 1`] = `
<path
  class="silky-path "
  d="M 10,30 A 20,20 0,0,1 50,30 A 20,20 0,0,1 90,30 Q 90,60 50,90 Q 10,60 10,30 z"
/>
`;

exports[`Path with Emotion 1`] = `
.emotion-0 {
  color: black;
}

<path
  class="silky-path emotion-0"
  d="M 10,30 A 20,20 0,0,1 50,30 A 20,20 0,0,1 90,30 Q 90,60 50,90 Q 10,60 10,30 z"
/>
`;

exports[`Path with custom class 1`] = `
<path
  class="silky-path foo"
  d="M 10,30 A 20,20 0,0,1 50,30 A 20,20 0,0,1 90,30 Q 90,60 50,90 Q 10,60 10,30 z"
/>
`;

exports[`Path with pathLength 1`] = `
<path
  class="silky-path "
  d="M 10,30 A 20,20 0,0,1 50,30 A 20,20 0,0,1 90,30 Q 90,60 50,90 Q 10,60 10,30 z"
  pathLength="2"
/>
`;
